libretroshare/src/pgp/pgphandler.cc

```diff
@@ -1381,10 +1381,12 @@ bool PGPHandler::decryptTextFromFile(const RsPgpId&,std::string& text,const std:
 
 bool PGPHandler::SignDataBin(const RsPgpId& id,const void *data, const uint32_t len, unsigned char *sign, unsigned int *signlen,bool use_raw_signature, std::string reason /* = "" */)
 {
-	RsStackMutex mtx(pgphandlerMtx) ;				// lock access to PGP memory structures.
 	// need to find the key and to decrypt it.
-	
-	const ops_keydata_t *key = locked_getSecretKey(id) ;
+	const ops_keydata_t *key = nullptr;
+	{
+		RS_STACK_MUTEX(pgphandlerMtx); // lock access to PGP memory structures.
+		key = locked_getSecretKey(id);
```

Comment:
If you really need that mutex here, then copy the struct entirely. It's costly, but that shouldn't be a problem here.

Reply:
Done